website/www/site/content/en/blog/dyi-content-discovery-platform-genai-beam.md

```diff
@@ -0,0 +1,327 @@
+
+*   A streaming pipeline that captures user content from wherever it resides.
+*   A process that extracts meaning from this content as a set of multi-dimensional vectors (text embeddings).
+*   A storage system that simplifies context matching between knowledge content and user inquiries (a Vector Database).
+*   Another storage system that maps knowledge representation with the actual content forming the aggregated context of the inquiry.
+*   A model capable of understanding the aggregated context and, through prompt engineering, delivering meaningful answers.
+*   HTTP and GRPC based services.
+
+These components work together to provide a comprehensive and simple implementation for a content discovery platform.
+
+## Architecture Design
+
+Given the multiple components in play, we will be diving down to explain how the different components interact to resolve the two main use cases of the platform.
+
+### Component’s Dependencies
+
+The following diagram shows all of the components that the platform integrates to capture documents for ingestion and resolve user query requests, also all the dependencies existing between the different components of the solution and the GCP services in use.
+
+<img class="center-block"
+    src="/images/blog/dyi-cdp-genai-beam/cdp-arch.png"
+    alt="Content Discovery Platform Interactions">
+
+As seen in the diagram, the context-extraction component is the central aspect in charge of retrieving the document’s content, also their semantic meaning from the embedding’s model and storing the relevant data (chunks text content, chunks embeddings, JSON-L content) in the persistent storage systems for later use. PubSub resources are the glue between the streaming pipeline and the asynchronous processing, capturing the user ingestion requests, retries from potential errors from the ingestion pipeline (like the cases on where documents have been sent for ingestion but the permission has not been granted yet, triggering a retry after some minutes) and content refresh events (periodically the pipeline will scan the ingested documents, review the latest editions and define if a content refresh should be triggered).
+
+Also, CloudRun plays the important role of exposing the services, which will interact with the many different storage systems in use to resolve the user query requests. For example, capturing the semantic meaning from the user’s query by interacting with the embedding’s model, finding near matches from the Vertex AI Vector Search (formerly Matching Engine),which stores the embedding vectors from the ingested document’s content, retrieving the text content from BigTable to contextualize finally the request to the VertexAI Text-Bison and Chat-Bison models for a final response to the user’s originary query.
```

Comment:
"to contextualize finally the request to the VertexAI Text-Bison and Chat-Bison models for a final response to the user’s originary query."

I don't follow what this is trying to say - is it just saying "to send the request to the VertexAI..."? I'd suggest rewording.

Also, I don't think we need to change this, but it is worth calling out that Beam has built in support for calling vertex AI endpoints (e.g. here's an example calling into a tuned vertex text-bison model - https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/vertex_ai_llm_text_classification.py). Maybe we could at least call out that integration point in this blog since I imagine it would be interesting to the audience

Reply:
Reworded the whole paragraph. 

This specific part talks about the interaction with vertex ai models from the context of the exposed services, not the Apache Beam pipeline, I will add a reference to this code example in the section with the pipeline details. Thanks!